beacon-chain/rpc/apimiddleware/custom_handlers.go

```diff
@@ -112,6 +119,53 @@ func handleGetSSZ(
 	return true
 }
 
+func handlePostSSZ(
+	m *apimiddleware.ApiProxyMiddleware,
+	endpoint apimiddleware.Endpoint,
+	w http.ResponseWriter,
+	req *http.Request,
+	config sszConfig,
+) (handled bool) {
+	if !sszPosted(req) {
+		return false
+	}
+
+	if errJson := prepareSSZRequestForProxying(m, endpoint, req, config.sszPath); errJson != nil {
+		apimiddleware.WriteError(w, errJson, nil)
+		return true
+	}
+	prepareCustomHeaders(req)
+	if errJson := preparePostedSSZData(req); errJson != nil {
+		apimiddleware.WriteError(w, errJson, nil)
+		return true
+	}
+
+	grpcResponse, errJson := m.ProxyRequest(req)
```

Comment:
Most of this logic below exists elsewhere right? Maybe we can deduplicate a little

Reply:
Deduplicating things make it look a bit ugly, I would leave the code as is.